source/agora/consensus/protocol/Nominator.d

```diff
@@ -769,6 +772,20 @@ extern(D):
         const cur_height = this.ledger.height();
         log.trace("Received BLOCK SIG {} from node {} for block {}",
                     block_sig.signature, block_sig.utxo, block_sig.height);
+
+        if (block_sig.height == pending_block.header.height)
+        {
+            if (this.collectBlockSignature(block_sig, pending_block.hashFull()))
+            {
+                log.dbg("{}: Add to next block #{} being externalized for utxo {}", __PRETTY_FUNCTION__, block_sig.height, block_sig.utxo);
+                log.dbg("{}: header was {}", __FUNCTION__, pending_block.header.validators);
+                this.updateMultiSignature(pending_block.header);
+                log.dbg("{}: header now {}", __FUNCTION__, pending_block.header.validators);
+            }
```

Comment:
The calls of the `collectBlockSignature` and `updateMultiSignature` are shown twice in this function. Can't we refactor the code like this which is not complete code?
```
if (block_sig.height == pending_block.header.height)
    block = pending_block;
else
    block = this.ledger.getBlocksFrom(Height(block_sig.height)).front;

if (this.collectBlockSignature(block_sig, block.hashFull()))
{
    log.dbg("{}: Add to next block #{} being externalized for utxo {}", __PRETTY_FUNCTION__, block_sig.height, block_sig.utxo);
    log.dbg("{}: header was {}", __FUNCTION__, block.header.validators);
    this.updateMultiSignature(block.header);
    log.dbg("{}: header now {}", __FUNCTION__, block.header.validators);
}
else
{
    return BlockHeight.init;
}

if (block_sig.height == pending_block.header.height)
    return BlockHeight.init;
```

Reply:
Refactored to be cleaner.